integration-libs/opf/cta/components/opf-cta-scripts/opf-cta-scripts.service.ts

```diff
@@ -99,84 +136,42 @@ export class OpfCtaScriptsService {
     paymentAccountIds: number[]
   ): Observable<CtaScriptsRequest> {
     if (!scriptsLocation) {
-      return throwError('Invalid Script Location');
+      return throwError(() => 'Invalid Script Location');
     }
-    const toBeImplementedException = () => throwError('to be implemented');
+    const toBeImplementedException = () =>
+      throwError(() => 'to be implemented');
     const locationToFunctionMap: Record<
       CtaScriptsLocation,
       () => Observable<CtaScriptsRequest>
     > = {
-      [CtaScriptsLocation.PDP_QUICK_BUY]: () =>
-        this.fillCtaRequestforPDP(scriptsLocation, paymentAccountIds),
       [CtaScriptsLocation.ORDER_HISTORY_PAYMENT_GUIDE]: () =>
-        this.fillCtaRequestforPagesWithOrder(scriptsLocation),
+        this.opfStaticCtaService.fillCtaRequestforPagesWithOrder(
+          scriptsLocation
+        ),
       [CtaScriptsLocation.ORDER_CONFIRMATION_PAYMENT_GUIDE]: () =>
-        this.fillCtaRequestforPagesWithOrder(scriptsLocation),
-      [CtaScriptsLocation.CART_MESSAGING]: toBeImplementedException,
+        this.opfStaticCtaService.fillCtaRequestforPagesWithOrder(
+          scriptsLocation
+        ),
+      [CtaScriptsLocation.CART_MESSAGING]: () =>
+        this.opfDynamicCtaService.fillCtaRequestforCartPage(
+          scriptsLocation,
+          paymentAccountIds
+        ),
+      [CtaScriptsLocation.PDP_MESSAGING]: () =>
+        this.opfDynamicCtaService.fillCtaRequestforProductPage(
+          scriptsLocation,
+          paymentAccountIds
+        ),
       [CtaScriptsLocation.CART_QUICK_BUY]: toBeImplementedException,
       [CtaScriptsLocation.CHECKOUT_QUICK_BUY]: toBeImplementedException,
-      [CtaScriptsLocation.PDP_MESSAGING]: toBeImplementedException,
+      [CtaScriptsLocation.PDP_QUICK_BUY]: toBeImplementedException,
```

Comment:
For info, PDP_QUICK_BUY and CART_QUICK_BUY are the more advanced CTA like standalone Paypal. not on scope yet.